src/v/archival/ntp_archiver_service.cc

```diff
@@ -647,6 +648,16 @@ ntp_archiver::download_manifest() {
 
 ss::future<cloud_storage::upload_result> ntp_archiver::upload_manifest(
   std::optional<std::reference_wrapper<retry_chain_node>> source_rtc) {
+    if (!_feature_table.local().is_active(
+          features::feature::cloud_log_size_in_manifest)) {
+        vlog(
+          archival_log.info,
+          "Skipping manifest upload until all nodes in the cluster have been "
+          "upgraded.");
+
+        co_return cloud_storage::upload_result::cancelled;
+    }
```

Comment:
In the current state of `dev`, this logic could lead to us getting stuck in certain situations. That shouldn't be the case after https://github.com/redpanda-data/redpanda/pull/9154 merges. When this branch is taken and the upload gets skipped, the manifest remains dirty, and the upload will be retried at a later point.